ruby/ql/lib/codeql/ruby/frameworks/Array.qll

```diff
@@ -0,0 +1,2407 @@
+  }
+
+  private class MinmaxNoArgNoBlockSummary extends MinmaxSummary {
+    MinmaxNoArgNoBlockSummary() {
+      this = "minmax_no_block" and
+      not exists(mc.getBlock())
+    }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "ArrayElement of Receiver" and
+      output = "ArrayElement[?] of ReturnValue" and
+      preservesValue = true
+    }
+  }
+
+  private class MinmaxBlockSummary extends MinmaxSummary {
+    MinmaxBlockSummary() {
+      this = "minmax_block" and
+      exists(mc.getBlock())
+    }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "ArrayElement of Receiver" and
+      output =
+        [
+          "Parameter[0] of BlockArgument", "Parameter[1] of BlockArgument",
+          "ArrayElement[?] of ReturnValue"
+        ] and
+      preservesValue = true
+    }
+  }
+
+  private class MinmaxBySummary extends SimpleSummarizedCallable {
+    MinmaxBySummary() { this = "minmax_by" }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "ArrayElement of Receiver" and
+      output = ["Parameter[0] of BlockArgument", "ArrayElement[?] of ReturnValue"] and
+      preservesValue = true
+    }
+  }
+
+  private class PartitionSummary extends SimpleSummarizedCallable {
+    PartitionSummary() { this = "partition" }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "ArrayElement of Receiver" and
+      output =
+        ["Parameter[0] of BlockArgument", "ArrayElement[?] of ArrayElement[?] of ReturnValue"] and
+      preservesValue = true
+    }
+  }
+
+  private class QuerySummary extends SimpleSummarizedCallable {
+    QuerySummary() { this = ["all?", "any?", "none?", "one?"] }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "ArrayElement of Receiver" and
+      output = "Parameter[0] of BlockArgument" and
+      preservesValue = true
+    }
+  }
+
+  private class RejectSummary extends SimpleSummarizedCallable {
+    RejectSummary() { this = "reject" }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "ArrayElement of Receiver" and
+      output = ["Parameter[0] of BlockArgument", "ArrayElement[?] of ReturnValue"] and
+      preservesValue = true
+    }
+  }
+
+  private class RejectBangSummary extends SimpleSummarizedCallable {
+    RejectBangSummary() { this = "reject!" }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "ArrayElement of Receiver" and
+      output =
+        [
+          "ArrayElement[?] of ReturnValue", "ArrayElement[?] of Receiver",
+          "Parameter[0] of BlockArgument"
+        ] and
+      preservesValue = true
+    }
```

Comment:
Confusingly, this is actually a method on `Array`, not `Enumerable`. I think maybe because it modifies the receiver? 🤷

Reply:
Good catch. I've moved it.